Source/Core/Core/CoreTiming.cpp

```diff
@@ -371,6 +371,11 @@ void CoreTimingManager::Throttle(const s64 target_cycle)
   const TimePoint min_deadline = time - max_fallback;
   const TimePoint max_deadline = time + max_fallback;
 
+  if (speed <= 0.0)
```

Comment:
This is kind of explained in the PR description already, but could you further explain what it is for?
Also, should we skip this branch if audio stretching is enabled?

Reply:
> Previously `m_throttle_deadline` was not being updated during fast-forward, causing the emulator to try to render as many frames as possible in zero seconds. And when you released fast-forward, the emulator would continue to fast-forward for an emulated time equal to the physical time you held the key, waiting for `m_throttle_deadline` to catch up to the physical time. This was only limited when the "System can not to keep up with timings!" fallback would keep the target timebase 100ms before the current time, so the emulator would never fast-forward more than 100ms emulated time after you released Tab.

In my opinion, the previous behavior of fast-forwarding the emulated game for up to 100ms of emulated time after releasing the fast-forward key, was a bug. And you should stop fast-forwarding (begin throttling emulation using the current host time) as soon as you release the fast-forward key, with or without audio stretching enabled. This change accomplishes it.

----

Though another route to take would be to neither write nor read `m_throttle_deadline` while fast-forwarding, and when transitioning from fast-forwarding to regular speed, assign `m_throttle_deadline = time` (either before or after `m_throttle_deadline += cycles...`) to begin throttling from that time (my current PR's code sleeps until the `time` of the final fast-forward plus the `cycles` of the current non-fast-forward emulation block). These approaches will have subtly different timing characteristics when ending fast-forward.

- It seems `CoreTimingManager::Advance()` calls throttle *before* running a computation at a given emulated/real time? **I'm still not sure what the `Advance()` doc comment and PowerPC downcount mean.**

Reassigning `m_throttle_deadline` *after* ending fast-forward, requires adding a variable to cache whether fast-forward was enabled or not on the previous Throttle call. I'm not sure if the result has more or less "correct" timing behavior, but realistically there's no difference if we advance timing thousands of times per second.

Adding a "was fast forwarding" variable adds code complexity, but the same variable can be used for "making CoreTimingManager inform the mixer when CoreTimingManager itself stops fast-forwarding" (another proposal).

Reply:
Downcount is a variable the JIT uses to keep track of how many emulated clock cycles are left until Advance needs to be called again. For instance, let's say that Advance is called and it decides that 2000 cycles are left until the next scheduled event. It will then set downcount to 2000 and return control to the JIT. As the JIT executes code, it will decrease the downcount variable little by little, and between each JIT block, if it notices the downcount is 0 or negative, it will call Advance again.